### PR TITLE
added recipe for seqan

### PR DESCRIPTION
**Pull request by boussaffawalid**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/seqan/2.x.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            tools.check_min_cppstd(self, 14)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "seqan-library-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.download(
+            url="https://raw.githubusercontent.com/seqan/seqan/master/LICENSE",
+            filename=os.path.join(self._source_subfolder, "LICENSE")
+        )
```
Comment (prince-chrismc) [inline recipes/seqan/2.x.x/conanfile.py:38]: you can avoid this is you download the sources from github https://github.com/seqan/seqan/archive/seqan-v2.4.0.tar.gz

just a thought, not a requirement

File: recipes/seqan/2.x.x/test_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    // gaps, this operation is symmetric.
+    std::cout << "3 clipped ungapped => " << 3 + beginPosition(gaps) << " unclipped ungapped\n"
+              << "5 unclipped ungapped => " << 5 - beginPosition(gaps) << " clipped ungapped\n\n";
+
+    // Translating between gapped clipped position and ungapped clipped
+    // position and between gapped unclipped and ungapped unclipped positions
+    // has to be done using the translations above.
+    std::cout << "3 clipped gapped => " << toSourcePosition(gaps, 3) - beginPosition(gaps) << " clipped ungapped\n"
+              << "4 unclipped ungapped => " << toViewPosition(gaps, 4) + clippedBeginPosition(gaps) << " unclipped gapped\n";
+
```
Comment (prince-chrismc) [inline recipes/seqan/2.x.x/test_package/test_package.cpp:69]: Please try to keep the test_package a minimum to ensure correctness of the package not a sample for the API

```suggestion
```

File: recipes/seqan/2.x.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import os
+from conans import ConanFile, tools
+
+required_conan_version = ">=1.28.0"
+
+
```
Comment (prince-chrismc) [inline recipes/seqan/2.x.x/conanfile.py:6]: There's no components so this is not requires

```suggestion
```

File: recipes/seqan/2.x.x/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} CONAN_PKG::seqan)
+#SeqAn requires C++14!
```
Comment (prince-chrismc) [inline recipes/seqan/2.x.x/test_package/CMakeLists.txt:9]:
```suggestion
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — COMMENTED** [review]
🚀 LGTM!

File: recipes/seqan/2.x.x/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
```
Comment (uilianries) [inline recipes/seqan/2.x.x/conanfile.py:26]: @property
    def _compilers_minimum_version(self):
        return {
            "gcc": "5",
            "Visual Studio": "14",
            "clang": "3.4",
            "apple-clang": "3.4",
        }

    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 14)
        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
        if minimum_version:
            if tools.Version(self.settings.compiler.version) < minimum_version:
                raise ConanInvalidConfiguration("seqan requires C++14, which your compiler does not fully support.")
        else:
            self.output.warn("seqan requires C++14. Your compiler is unknown. Assuming it supports C++14.")

File: recipes/seqan/2.x.x/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import os
+from conans import ConanFile, tools
```
Comment (uilianries) [inline recipes/seqan/2.x.x/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
from conans.errors import ConanInvalidConfiguration
```

**Review by uilianries — CHANGES_REQUESTED** [review]
Have to filter compiler version according cppstd support.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/seqan/2.x.x/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+from conans.errors import ConanInvalidConfiguration
+
+class SeqanConan(ConanFile):
+    name = "seqan"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seqan/seqan"
+    description = """
+SeqAn is an open source C++ library of efficient algorithms and data structures for the analysis of sequences
```
Comment (SSE4) [inline recipes/seqan/2.x.x/conanfile.py:10]: I'd personally just leave the first sentence. but never mind :)

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**